prow/pod-utils/clone/clone.go

```diff
@@ -180,13 +180,8 @@ func (g *gitCtx) commandsForBaseRef(refs prowapi.Refs, gitUserName, gitUserEmail
 	if gitUserEmail != "" {
 		commands = append(commands, g.gitCommand("config", "user.email", gitUserEmail))
 	}
-	if cookiePath != "" {
-		if refs.SkipSubmodules {
-			commands = append(commands, g.gitCommand("config", "http.cookiefile", cookiePath))
-		} else {
-			// --global to ensure that all submodules can use this auth
-			commands = append(commands, g.gitCommand("config", "--global", "http.cookiefile", cookiePath))
-		}
+	if cookiePath != "" && refs.SkipSubmodules {
+		commands = append(commands, g.gitCommand("config", "http.cookiefile", cookiePath))
```

Comment:
Why do we need to configure this locally when skipping submodules? I don't see how that changes the behavior compared to always configuring globally, especially in the case where multiple refs cause us to configure the cookiefile both globally and locally.

Reply:
If we are not using submodules, we do not need to overwrite the global config.

The local config takes precedence over the global one, and is less likely to have side effects.